File: challenges/houseNumbersSum/python3/houseNumbersSum.py
def houseNumbersSum(a):
	s = 0
	for x in a:
		if x == 0:
			break
		else:
			s += x
	return s

if __name__ == '__main__':
	input0 = [[5, 1, 2, 3, 0, 1, 5, 0, 2], [4, 2, 1, 6, 0], [4, 1, 2, 3, 0, 10, 2], [0, 1, 2, 3, 0], [5, 1, 1, 3, 0, 1, 5, 0, 2], [10, 10, 10, 10, 10, 10, 10, 10, 10, 0], [10, 10, 10, 10, 10, 10, 10, 10, 0, 10]]
	expectedOutput = [11, 13, 10, 0, 10, 90, 80]
	assert len(input0) == len(expectedOutput), '# input0 = {}, # expectedOutput = {}'.format(len(input0), len(expectedOutput))
	for i, expected in enumerate(expectedOutput):
		actual = houseNumbersSum(input0[i])
		assert actual == expected, 'houseNumbersSum({}) returned {}, but expected {}'.format(input0[i], actual, expected)
	print('PASSES {} out of {} tests'.format(len(expectedOutput), len(expectedOutput)))

File: challenges/houseNumbersSum/octave/houseNumbersSum.m
function r = houseNumbersSum(a)
	r = 0;
	for x = a
		if x == 0
			break;
		endif
		r = r + x;
	endfor
endfunction
%!assert(houseNumbersSum([5, 1, 2, 3, 0, 1, 5, 0, 2]), 11)
%!assert(houseNumbersSum([4, 2, 1, 6, 0]), 13)
%!assert(houseNumbersSum([4, 1, 2, 3, 0, 10, 2]), 10)
%!assert(houseNumbersSum([0, 1, 2, 3, 0]), 0)
%!assert(houseNumbersSum([5, 1, 1, 3, 0, 1, 5, 0, 2]), 10)
%!assert(houseNumbersSum([10, 10, 10, 10, 10, 10, 10, 10, 10, 0]), 90)
%!assert(houseNumbersSum([10, 10, 10, 10, 10, 10, 10, 10, 0, 10]), 80)